File: Algebra_MultiplyFraction.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace Algebra_CS
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Welcome to our demonstration sequels");
            Console.WriteLine("Hope you enjoy (and follow) the lessons.");
            Console.WriteLine("\r\n");

            /*
            * Multiplying fractions
             */
            List<int> numerators = new List<int>();
            numerators.Add(16);
            numerators.Add(20);
            numerators.Add(27);
            numerators.Add(20);

            List<int> denominators = new List<int>();
            denominators.Add(9);
            denominators.Add(9);
            denominators.Add(640);
            denominators.Add(7);

            Console.WriteLine("    Solving:");
            // Print as fraction
            foreach (int n in numerators)
            {
                Console.Write(String.Format("{0,13}", n));
            }
            Console.Write(Environment.NewLine + String.Format("{0,12}", " "));
            for (int i = 0; i < numerators.Count - 1; i++)
            {
                Console.Write(String.Format("{0}", "-     X      "));
            }
            Console.WriteLine(String.Format("{0,1}", "-"));
            foreach (int d in denominators)
            {
                Console.Write(String.Format("{0,13}", d));
            }
            Console.WriteLine();

            // use the MultiplyFraction class
            MultiplyFraction mul_fract = new MultiplyFraction(numerators, denominators);
            int[] solution = mul_fract.doMultiply();

            Console.WriteLine(Environment.NewLine);

            Console.WriteLine(String.Format("{0,25}", solution[0]));
            Console.WriteLine(String.Format("{0,25}", "Answer   =   -"));
            Console.WriteLine(String.Format("{0,25}", solution[1]));

        }
    }
}

File: Algebra_MultiplyFraction.pl
#!/usr/bin/perl;
use strict;
use warnings;
use MULTIPLYFRACTION;

# Useful variables
my (@numerators, @denominators);
my (%fraction, %fractions);

##
 # Multiplying fractions
 ##
@numerators	  = (16, 20, 27, 20);
@denominators = (9, 9, 640, 7);
%fractions = (
	numerators	 => \@numerators,
	denominators => \@denominators
);

print "\n    Solving:\n";
# Print as fraction
printf("%13u", $_) for @numerators;
printf("\n%12s", " ");
print "-     X      " for 1 .. $#numerators;
printf("%1s", "-");
print "\n";
printf("%13u", $_) for @denominators;
print "\n";

# use the MultiplyFraction class
my $mul_fract = MULTIPLYFRACTION->new(\%fractions);
%fraction = %{$mul_fract->doMultiply()};

printf("\n%25u\n", $fraction{numerator});
printf("%25s\n", "Answer =      -");
printf("%25u\n", $fraction{denominator});


print "\n\n";